test/unit/preact/test-base-element.js

```diff
@@ -0,0 +1,442 @@
+  });
+
+  function waitFor(callback, errorMessage) {
+    return poll(
+      errorMessage,
+      () => {
+        return callback();
+      },
+      undefined /** opt_onError */,
+      200 /** opt_timeout */
+    );
+  }
+
+  function waitForMutation(element, callback) {
+    return new Promise((resolve) => {
+      const mo = new MutationObserver(() => {
+        resolve();
+        mo.disconnect();
+      });
+      mo.observe(element, {
+        attributes: true,
+        childList: true,
+        subtree: true,
+        characterData: true,
+      });
+      callback();
+    });
+  }
+
+  describe('no children mapping', () => {
+    let element;
+
+    beforeEach(async () => {
+      Impl['props'] = {
+        'noValue': {attr: 'no-value'},
+        'valueWithDef': {attr: 'value-with-def', default: 'DEFAULT'},
+        'propA': {attr: 'prop-a'},
+        'minFontSize': {attr: 'min-font-size', type: 'number'},
+        'disabled': {attr: 'disabled', type: 'boolean'},
+        'enabled': {attr: 'enabled', type: 'boolean'},
+      };
+      element = html`
+        <amp-preact layout="fixed" width="100" height="100"
+            prop-a="A"
+            min-font-size="72"
+            disabled
+            unknown="1">
+        </amp-preact>
+      `;
+      doc.body.appendChild(element);
+      await element.build();
+      await waitFor(() => component.callCount > 0, 'component rendered');
+    });
+
+    it('should render content inline', async () => {
+      expect(component).to.be.calledOnce;
+      expect(element.querySelector('#component')).to.be.ok;
+      expect(element.shadowRoot).to.not.be.ok;
+    });
+
+    it('should parse attributes on first render', async () => {
+      expect(component).to.be.calledOnce;
+      expect(lastProps).to.deep.equal({
+        valueWithDef: 'DEFAULT',
+        propA: 'A',
+        minFontSize: 72,
+        disabled: true,
+        enabled: false,
+      });
+    });
+
+    it('should mutate attributes', async () => {
+      element.setAttribute('prop-a', 'B');
+      element.setAttribute('min-font-size', '72.5');
+      element.setAttribute('enabled', '');
+      element.removeAttribute('disabled');
+
+      await waitFor(() => component.callCount > 1, 'component re-rendered');
+
+      expect(component).to.be.calledTwice;
+      expect(lastProps).to.deep.equal({
+        valueWithDef: 'DEFAULT',
+        propA: 'B',
+        minFontSize: 72.5,
+        disabled: false,
+        enabled: true,
+      });
+    });
+
+    it('should ignore non-declared attributes', async () => {
+      await waitForMutation(element, () => {
+        element.removeAttribute('unknown');
+        element.setAttribute('unknown2', '2');
+        element.style.background = 'red';
+        element.classList.add('class1');
+      });
+
+      // Skip couple more animation frames.
+      await new Promise(resolve => setTimeout(resolve, 32));
+
+      expect(component).to.be.calledOnce;
```

Comment:
From bento chat on how to check for false-positive.  Would the following work?  Then we can be slightly more confident that we only have one more render and it's happening from updating 'prop-a'

`element.setAttribute('prop-a', 'B');
await waitFor(() => component.callCount > 1, 'component re-rendered');
expect(component).to.be.calledTwice;`

Reply:
Good idea. Done.